### PR TITLE
Clean up logic for deprecated flag --container-runtime in kubelet

### PR DESCRIPTION
**Pull request by hakman**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
As part of the removal of dockershim and deprecation of the `--container-runtime` flag, the logic behind it should also be removed from kubelet.

#### Which issue(s) this PR fixes:

Refs: #106893

#### Special notes for your reviewer:
Refs: #107094

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/gce/config-default.sh
Comment (adisky) [inline cluster/gce/config-default.sh:109]: we might not want to remove the if condition with containerd, as this script might be used with some other container runtimes, not fully sure on this but there is no harm in keeping it as well
```
if [[ "${CONTAINER_RUNTIME}" == "containerd"
```

this is really weird is `CONTAINER_RUNTIME` here not same as --container-runtime kubelet flag? because kubelet flag cannot be set to `containerd`

**Review by adisky — COMMENTED** [review]
Also this PR getting large to review, can we separate the into different commits/PR one for cluster/gce scripts and one for kubelet code changes?

File: cluster/gce/config-default.sh
Comment (hakman) [reply cluster/gce/config-default.sh:109]: The new PR for scripts is https://github.com/kubernetes/kubernetes/pull/107663.

The `"${CONTAINER_RUNTIME}" == "containerd"` was mistake done at some point. This is why this hack was needed:
https://github.com/kubernetes/kubernetes/blob/94ff5ea1f1727826c9848878c890aa793f733aec/test/e2e/framework/test_context.go#L489-L493

If anyone wants to use a different CRI, should be enough to set `KUBE_CONTAINER_RUNTIME_NAME`, `KUBE_CONTAINER_RUNTIME_ENDPOINT` and  `KUBE_LOAD_IMAGE_COMMAND`.

Comment (adisky) [reply cluster/gce/config-default.sh]: Thanks for the explantation @hakman sounds good!!

File: cmd/kubelet/app/options/options.go
Comment (endocrimes) [inline cmd/kubelet/app/options/options.go:195]:
```suggestion
		return fmt.Errorf("unsupported CRI runtime: %q, only %q is currently supported", f.ContainerRuntime, kubetypes.RemoteContainerRuntime)
```

this would potentially help folks move forward if they are stuck.

File: cmd/kubelet/app/server.go
Comment (endocrimes) [inline cmd/kubelet/app/server.go:189]:
```suggestion
				klog.InfoS("--pod-infra-container-image will not be pruned by the image garbage collector in kubelet and should also be set in the remote runtime")
```

File: pkg/kubelet/kubelet_pods.go
Comment (endocrimes) [inline pkg/kubelet/kubelet_pods.go:151]: Update the comments? - Removing supportsSingleFileMapping removes the formal containerd constraint from this.

**Review by endocrimes — COMMENTED** [review]
took a quick first pass, looking mostly good. Will review again when I'm back full time.

**Comment by hakman** [comment]
> took a quick first pass, looking mostly good. Will review again when I'm back full time.

Thanks @endocrimes :).

**Review by adisky — COMMENTED** [review]
/lgtm

**Review by endocrimes — APPROVED** [review]
/lgtm

**Comment by hakman** [comment]
Thanks @endocrimes. Should I squash the second commit?

**Comment by endocrimes** [comment]
@hakman feel free too and i'll lgtm again :)

**Comment by hakman** [comment]
> @hakman feel free too and i'll lgtm again :)

Done. Thanks @endocrimes. :)

**Review by yanghesong — APPROVED** [review]

**Review by endocrimes — APPROVED** [review]
/lgtm

For kubelet:
/assign @random-liu @mrunalp 

For kubemark:
/assign @shyamjvs

**Comment by adisky** [comment]
@hakman can you rebase it?

**Comment by hakman** [comment]
> @hakman can you rebase it?

Sure, done. Could we get someone to do the final review for this?

**Review by adisky — COMMENTED** [review]
/lgtm
/assign @dims @dchen1107 @derekwaynecarr 
(for approval)

**Comment by dims** [comment]
/approve
/hold

@hakman please give it a bit of time for sig-node approvers to peek (say end of the week?) and then remove hold

**Comment by hakman** [comment]
Sure. Thanks @dims!

**Review by endocrimes — APPROVED** [review]
/lgtm

the timer has also now expired and this can be merged.
/unhold

**Pull request merged**
